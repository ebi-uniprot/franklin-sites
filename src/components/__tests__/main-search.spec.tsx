import { screen, render, fireEvent } from '@testing-library/react';

import MainSearch from '../main-search';

describe('MainSearch component', () => {
  const handleSubmit = jest.fn();
  const handleChange = jest.fn();
  const props = {
    onTextChange: handleChange,
    onSubmit: handleSubmit,
  };
  const namespaces = { one: 'One', two: 'Two', three: 'Three' };

  it('should render', () => {
    const { asFragment } = render(<MainSearch {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with namespaces selector', () => {
    const { asFragment } = render(
      <MainSearch
        {...props}
        namespaces={namespaces}
        selectedNamespace="one"
        onNamespaceChange={jest.fn()}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should submit the search term', () => {
    render(<MainSearch {...props} />);
    fireEvent.submit(screen.getByRole('form'));
    expect(handleSubmit).toHaveBeenCalled();
  });

  it('should detect the search term', () => {
    render(<MainSearch {...props} />);
    fireEvent.change(screen.getByRole('searchbox'), {
      target: { value: 'foo' },
    });
    expect(handleChange).toHaveBeenCalledWith('foo');
  });

  it('should set searchTerm', () => {
    render(<MainSearch {...props} searchTerm="blah" />);
    expect((screen.getByRole('searchbox') as HTMLInputElement).value).toBe(
      'blah'
    );
  });
});
