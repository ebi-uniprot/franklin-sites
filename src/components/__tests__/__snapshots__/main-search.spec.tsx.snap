// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainSearch component should render 1`] = `
<DocumentFragment>
  <form
    aria-label="Main search"
    class="main-search"
  >
    <div
      class="main-search__input-container"
    >
      <input
        aria-label="Text query"
        type="text"
        value=""
      />
    </div>
    <button
      class="button primary"
      type="submit"
    >
      Search
    </button>
  </form>
</DocumentFragment>
`;

exports[`MainSearch component should render with namespaces selector 1`] = `
<DocumentFragment>
  <form
    aria-label="Main search"
    class="main-search"
  >
    <div
      aria-expanded="false"
      aria-haspopup="true"
      class="dropdown"
    >
      <button
        class="button primary"
        style="--main-button-color: #00639a;"
        type="button"
      >
        One
      </button>
    </div>
    <div
      class="main-search__input-container"
    >
      <input
        aria-label="Text query in one"
        type="text"
        value=""
      />
    </div>
    <button
      class="button primary"
      type="submit"
    >
      Search
    </button>
  </form>
</DocumentFragment>
`;
